prog/hints_test.go

```diff
@@ -34,31 +36,111 @@ func TestHintsCheckConstArg(t *testing.T) {
 	t.Parallel()
 	var tests = []ConstArgTest{
 		{
-			"One replacer test",
-			0xdeadbeef,
-			CompMap{0xdeadbeef: uint64Set{0xdeadbeef: true, 0xcafebabe: true}},
-			[]uint64{0xcafebabe},
+			name:  "One replacer test",
+			in:    0xdeadbeef,
+			size:  4,
+			comps: CompMap{0xdeadbeef: uint64Set{0xdeadbeef: true, 0xcafebabe: true}},
+			res:   []uint64{0xcafebabe},
 		},
 		// Test for cases when there's multiple comparisons (op1, op2), (op1, op3), ...
 		// Checks that for every such operand a program is generated.
 		{
-			"Multiple replacers test",
-			0xabcd,
-			CompMap{0xabcd: uint64Set{0x2: true, 0x3: true}},
-			[]uint64{0x2, 0x3},
+			name:  "Multiple replacers test",
+			in:    0xabcd,
+			size:  2,
+			comps: CompMap{0xabcd: uint64Set{0x2: true, 0x3: true}},
+			res:   []uint64{0x2, 0x3},
 		},
 		// Checks that special ints are not used.
 		{
-			"Special ints test",
-			0xabcd,
-			CompMap{0xabcd: uint64Set{0x1: true, 0x2: true}},
-			[]uint64{0x2},
+			name:  "Special ints test",
+			in:    0xabcd,
+			size:  2,
+			comps: CompMap{0xabcd: uint64Set{0x1: true, 0x2: true}},
+			res:   []uint64{0x2},
+		},
+
+		// The following tests check the size limits for each replacer and for the initial value
+		// of the argument. The checks are made for positive and negative values and also for bitfields.
+		{
+			name: "Int8 invalid value (positive)",
+			in:   0x1234,
+			size: 1,
+			comps: CompMap{
+				// void test8(i8 el) {
+				//		i16 w = (i16) el
+				//		if (w == 0x88) {...}
+				//		i16 other = 0xfffe
+				// 		if (w == other)
+				//  }; test8(i8(0x1234));
+				0x34: uint64Set{0x88: true, 0x1122: true, 0xfffffffffffffffe: true},
+				// This key should be iggnored.
+				0x1234: uint64Set{0xa1: true},
```

Comment:
Let's also add case with first argument 0xffffffffffffff34. It should also be ignored.